overlord/state/task.go

```diff
@@ -200,6 +200,41 @@ func (t *Task) Summary() string {
 }
 
 // Status returns the current task status.
+//
+// Possible state transitions:
+//
+//       /--aborting lane--Do
+//       V                 |
+//     >Hold               V
+//     |                 Doing-->Wait
+//     |                /  |  \
+//     |         abort /   V   V
+//   no undo          /  Done  Error
+//     |             V     |
+//     \----------Abort   aborting lane
```

Comment:
Bit of a nitpick, feel free to ignore. Initially I missed the sideways arrow left of Hold and was a bit confused, it might be slightly clearer with an upwards one in its own line. Added a vertical bar and some spaces for symmetry as well

```suggestion
//       /--aborting lane--Do
//       |                 |
//       V                 V
//     Hold              Doing-->Wait
//      ^                /  |  \
//      |         abort /   V   V
//   no undo           /  Done  Error
//      |             V     |
//      \----------Abort   aborting lane
```